### PR TITLE
add metadata to isRoom

### LINKED ISSUE #58: Room metadata when peer calls isRoom()
**Actual response:** `boolean`

**Expected response:**

```
{
  name: 'hermies.club', // i.e. the room's host
  features: [
    'tunnel', 
    'room1',
    // 'room2', // once we have attendants() and metadata()
    'alias',
    'httpAuth',
    'httpInvite'
  ]
}
```

-------

From Wouter's [UX research](https://www.manyver.se/ux-research/):

> Another note for a server feature: in the design of the Join room server screen, I conceptualised a new feature for room servers. Where a room server admin could link their personal SSB account to the server. This is expected to increase trust for potential new users of this room server, because they can then see which person owns it. (See the notes in report 3)

Actually, room1 answers `rpc.tunnel.isRoom` with an object which gives metadata about the room:

https://github.com/staltz/ssb-room/blob/e78d54887682664def36d48ca9e648fc609478e9/tunnel/server.js#L71-L77

For room2, we could support the same, while also adding more metadata, such as the admin's feed id, and the moderators feed ids.

### PR DESCRIPTION
**Pull request by cryptix**
Fixes #58

### CODE REVIEW

File: muxrpc/handlers/tunnel/server/state.go [hunk 1]
```diff
@@ -5,30 +5,74 @@ package server
...
+
+	var reply IsRoomReply
+	reply.Name = h.netInfo.Domain
+
+	// check if caller is a member
+	if _, err := h.members.GetByFeed(ctx, *ref); err != nil {
+		if errors.Is(err, roomdb.ErrNotFound) {
+			reply.Membership = false
+		}
+		return nil, err
```
Comment (staltz) [inline muxrpc/handlers/tunnel/server/state.go:55]: Don't we need an `else` here?

Comment (cryptix) [reply muxrpc/handlers/tunnel/server/state.go]: It was a bit convoluted, i think and hope the new version is clearer.

Comment (staltz) [reply muxrpc/handlers/tunnel/server/state.go]: looks good!

**Comment by cryptix** [comment]
@staltz if you have a JS module to check the format, we could add a test using that, too.

Elsewise this is good to go, i think.

**Review by staltz — APPROVED** [review]
IMO ready to go! Regarding tests, we could use ssb-room-client which already has some rudimentary tests for these metadata: https://github.com/ssb-ngi-pointer/ssb-room-client/blob/02b54773c78a9bb4fe77b933777bb1eb5a3bf33b/test/success-cases.js#L83-L88

Something like 

- go-ssb-room creates invite
- Node.js client accepts an invite from the Go server using ssb-http-invite-client
- Node.js client connects to the go-ssb-room
- isRoom happens in the background and Node.js client saves data in conn.json
- We check what was written to conn.json

But this can be done separately from this PR, if you're looking forward to press this green button now.

**Comment by cryptix** [comment]
I think my prio no1 is to unblock you with the stuff you need for the community launch and so will take a spin on the go side of #110 next instead of this integration test.

**Pull request merged**